internal/web_api.go

```diff
@@ -162,6 +180,13 @@ func (a *fileUploadAPI) UploadFile(filename string, file []byte) (string, error)
 		URL string `json:"url"`
 	}
 	err = a.send(req, &resp)
+
+	executionTime := time.Now().Sub(start)
+	if err != nil {
+		a.s.sample(metrics.APICallStats{"upload_file", executionTime, false})
+	} else {
+		a.s.sample(metrics.APICallStats{"upload_file", executionTime, true})
+	}
```

Comment:
oneline: `a.s.sample(metrics.APICallStats{"upload_file", executionTime, err == nil})`

Reply:
#crit check if channel is even set :D